hepdata_validator/data_file_validator.py

```diff
@@ -223,17 +248,37 @@ def check_length_values(self, file_path, data):
 
         :param data: data table in YAML format
         """
-        if 'independent_variables' in data and 'dependent_variables' in data and \
-              data['independent_variables'] is not None and data['dependent_variables'] is not None:
-            indep_count = [len(indep['values']) for indep in data['independent_variables'] if 'values' in indep]
-            dep_count = [len(dep['values']) for dep in data['dependent_variables'] if 'values' in dep]
-            if len(set(indep_count + dep_count)) > 1:  # if more than one unique count
-                error = ValidationError(
-                    "Inconsistent length of 'values' list: " +
-                    "independent_variables %s, dependent_variables %s" % (str(indep_count), str(dep_count)),
-                    instance=data
+        indep_count = [len(indep['values']) for indep in data['independent_variables'] if 'values' in indep]
+        dep_count = [len(dep['values']) for dep in data['dependent_variables'] if 'values' in dep]
+        if len(set(indep_count + dep_count)) > 1:  # if more than one unique count
+            error = ValidationError(
+                "Inconsistent length of 'values' list: " +
+                "independent_variables %s, dependent_variables %s" % (str(indep_count), str(dep_count)),
+                instance=data
+            )
+            self.add_validation_error(file_path, error)
+
+    def convert_to_float(self, error, file_path, path, instance):
+        """
+        Convert error from a string to a float if possible.
+
+        :param error: uncertainty from either 'symerror' or 'asymerror'
+        :return: error as a float if possible, otherwise the original string
+        """
+        if isinstance(error, str):
+            error = error.replace('%', '')  # strip percentage symbol
+        try:
+            error = float(error)
+        except ValueError:
+            if error != '':  # empty string is allowed in some circumstances
+                validation_error = ValidationError(
+                    f"Invalid error value {error}: value must be a float (with or without %) or empty string",
```

Comment:
Maybe change the message after the colon to "value must be a number (possibly ending in %)", since the empty-string case is not allowed in general, only for the `asymerror` case for one of `plus` or `minus`.